api/external_approval.go

```diff
@@ -6,6 +6,20 @@ type ExternalApprovalType string
 // ExternalApprovalTypeFeishu is the ExternalApproval from feishu.
 const ExternalApprovalTypeFeishu = "bb.plugin.app.feishu"
 
+// ExternalApprovalCancelReason describes why the external approval is canceled by Bytebase.
+type ExternalApprovalCancelReason string
+
+const (
+	// ExternalApprovalCancelReasonGeneral is the general reason, used as a default.
```

Comment:
These don't seem to be enums. We can just use strings instead of introducing a new type ExternalApprovalCancelReason.